src/libs/migrations/ReformatTimezone.js

```diff
@@ -0,0 +1,41 @@
+import Onyx from 'react-native-onyx';
+import _ from 'underscore';
+import ONYXKEYS from '../../ONYXKEYS';
+import CONST from '../../CONST';
+
+// This migration changes the format of the timezone in the Onyx key MY_PERSONAL_DETAILS from a string to an object
+export default function () {
+    return new Promise((resolve) => {
+        // Connect to the old key in Onyx to get the old value of myPersonalDetails timezone
+        // then update the timezone to be the default timezone and set the myPersonalDetails
+        // key with the updated values
+        const connectionID = Onyx.connect({
+            key: 'myPersonalDetails',
+            callback: (myPersonalDetails) => {
+                Onyx.disconnect(connectionID);
+
+                if (_.isUndefined(myPersonalDetails) || _.isEmpty(myPersonalDetails)) {
+                    console.debug('[Migrate Onyx] Skipped migration ReformatTimezone: No myPersonalDetails key found');
+                    return resolve();
+                }
+
+                // Fail early here because there is nothing to migrate, the timezone is already in the expected format
+                if (_.isObject(myPersonalDetails.timezone)) {
+                    console.debug('[Migrate Onyx] Skipped migration ReformatTimezone');
+                    return resolve();
+                }
+
+                // Replace the old timezone with the default timezone
```

Comment:
Hey @NikkiWines! I'm confused as to why we're setting the old string timezone to the DEFAULT_TIME_ZONE when migrating?

My current understanding is this: the migration isn't supposed to actually change the timezone, it's just meant to change its format so that when you pick it up in `DateUtil.js` you can be sure that it's an object (to prevent all that conditional logic you had before when reading it from Onyx). Why would we default it to `America/Los Angeles` here?

Reply:
This is a good point. I opted to use the default because we technically only have half the user's timezone NVP information if it's in string format. We'd be missing the `automatic: true/false` portion and using the default value instead just seemed like a simple solution rather than trying to guess half of their NVP value. 

However, in retrospect,  I think it's fine to use the old `timezone` value for `timezone.selected` and default to true, since the user can always modify that in their settings in E.cash.